.github/workflows/build.yml

```diff
@@ -0,0 +1,271 @@
+        with:
+          feature: ${{ env.EXPERIMENTAL_JAVA }}
+      - uses: actions/setup-java@v2
+        with:
+          java-version: ${{ env.JDK_VERSION }}
+          distribution: jdkfile
+          jdkFile: ${{ env.JDK_FILE }}
+      - name: Prepare JDK_EXPERIMENTAL env var
+        shell: bash
+        run: echo "JDK_EXPERIMENTAL=$JAVA_HOME" >> $GITHUB_ENV
+      - uses: actions/setup-java@v2
+        with:
+          java-version: 16
+          distribution: adopt
+      - name: Cache Gradle
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle/caches/
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: ${{ runner.os }}-gradle-
+      - name: Gradle toolchains
+        uses: eskatos/gradle-command-action@v1
+        with:
+          gradle-version: rc
+          arguments: -Porg.gradle.java.installations.auto-download=false javaToolchains 
+      - name: Gradle build
+        uses: eskatos/gradle-command-action@v1
+        with:
+          gradle-version: rc
+          arguments: --refresh-dependencies -PexperimentalJavaVersion=17 -PmodularBuild=${{ matrix.modular }} -PjunitVersion=${{ matrix.junit-version }} -Porg.gradle.java.installations.auto-download=false --stacktrace --scan clean build
+
+  # Releasing will be only done, if there is a version defined and `releasing` is set to true.
+  # If not this stage will be ignored.
+  release:
+    if: github.event.inputs.version != '' && github.event.inputs.releasing == 'true'
+    runs-on: ubuntu-latest
+    needs: full-featured
+    timeout-minutes: 15
+    name: Release Pioneer into the wild
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Set up java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: Perform release
+        uses: eskatos/gradle-command-action@v1
+        env:
+          # used to trigger website uild
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # used by `io.github.gradle-nexus.publish-plugin` to sign artifacts
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.JUNIT_PIONEER_PGP_SECRET_KEY }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.JUNIT_PIONEER_PGP_PASSWORD }}
+          # used by `io.github.gradle-nexus.publish-plugin` to publish to Maven Central
+          ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.NEXUS_TOKEN_USERNAME }}
+          ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.NEXUS_TOKEN_PASSWORD }}
+          # defines released version according to GitHub Action input
+          ORG_GRADLE_PROJECT_version: ${{ github.event.inputs.version }}
+        with:
+          arguments: publishToSonatype closeAndReleaseSonatypeStagingRepository githubRelease
+
+  # After our release, we also need to trigger an update for our website build.
+  updateWebsite:
+    runs-on: ubuntu-latest
+    needs: release
+    timeout-minutes: 15
```

Comment:
All timeouts seem to be 15 min, maybe define a envvar `DEFAULT_JOB_TIMEOUT`?

Reply:
that is a good idea - i will add this as soon as i have time

Reply:
I tried, it lead to the build not starting...

Reply:
Thx for trying it out, didn't know/expect that.

As pointed out by @aepfli during the stream, I believe we can use [`defaults`](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaults), "[a] map of default settings that will apply to all jobs in the workflow".

```yaml
defaults:
  timeout-in-minutes: 15
```

This way, we should be able to remove the timeouts from all jobs.

Reply:
See 7304aae.